crates/token/src/lib.rs

```diff
@@ -82,127 +82,24 @@ where
     Serialize,
     Deserialize,
 )]
-pub struct TransparentTransfer(pub Vec<TransparentTransferData>);
-
-/// Arguments for a transparent token transfer
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    BorshSerialize,
-    BorshDeserialize,
-    BorshDeserializer,
-    BorshSchema,
-    Hash,
-    Eq,
-    PartialOrd,
-    Serialize,
-    Deserialize,
-)]
-pub struct TransparentTransferData {
-    /// Source address will spend the tokens
-    pub source: Address,
-    /// Target address will receive the tokens
-    pub target: Address,
-    /// Token's address
-    pub token: Address,
-    /// The amount of tokens
-    pub amount: DenominatedAmount,
-}
-
-/// Arguments for a shielded token transfer
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    BorshSerialize,
-    BorshDeserialize,
-    BorshDeserializer,
-    BorshSchema,
-    Hash,
-    Eq,
-    PartialOrd,
-    Serialize,
-    Deserialize,
-)]
-pub struct ShieldedTransfer {
-    /// Hash of tx section that contains the MASP transaction
-    pub section_hash: Hash,
-}
-
-/// Arguments for a shielding transfer (from a transparent token to a shielded
-/// token)
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    BorshSerialize,
-    BorshDeserialize,
-    BorshDeserializer,
-    BorshSchema,
-    Hash,
-    Eq,
-    PartialOrd,
-    Serialize,
-    Deserialize,
-)]
-pub struct ShieldingTransferData {
-    /// Source address will spend the tokens
-    pub source: Address,
-    /// Token's address
-    pub token: Address,
-    /// The amount of tokens
-    pub amount: DenominatedAmount,
-}
-
-/// Arguments for a shielding transfer (from a transparent token to a shielded
-/// token)
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    BorshSerialize,
-    BorshDeserialize,
-    BorshDeserializer,
-    BorshSchema,
-    Hash,
-    Eq,
-    PartialOrd,
-    Serialize,
-    Deserialize,
-)]
-pub struct ShieldingTransfer {
+pub struct Transfer {
     /// Transfer-specific data
-    pub data: ShieldingTransferData,
+    pub data: Vec<TransferData>,
```

Comment:
let's make this slightly more descriptive
```suggestion
    pub transparent: Vec<TransparentTransfer>,
```

Reply:
Done, thanks.